### PR TITLE
Feature 17 milestone list and create

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

File: src/components/milestones/new-milestone-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    await createMilestoneMutation.mutateAsync({
+      projectId: projectId,
+      milestone: {
+        name: milestoneData.name,
+        startDate: milestoneData.startDate.toJSDate(),
+        endDate: milestoneData.endDate.toJSDate(),
+      },
+    });
+
```
Comment (Daniilmass) [inline src/components/milestones/new-milestone-dialog.tsx:72]: Consider adding loader enabling/disabling

File: src/components/milestones/new-milestone-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    setMilestoneData({ name: "", startDate: null, endDate: null });
+    setOpen(false);
+  };
+
+  const isDisabled = !(!!milestoneData.name && !!milestoneData.startDate && !!milestoneData.endDate);
```
Comment (Daniilmass) [inline src/components/milestones/new-milestone-dialog.tsx:77]: Missing TSdocs

File: src/components/milestones/new-milestone-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  return (
```
Comment (Daniilmass) [inline src/components/milestones/new-milestone-dialog.tsx:79]: Missing TSdocs i.e. "Main component render"

File: src/hooks/api-queries.ts [hunk 4]
```diff
@@ -118,3 +126,22 @@ export const useListFilesQuery = () =>
     queryKey: ["files"],
     queryFn: () => filesApi.listFiles().catch(handleErrorWithMessage("Error listing files")),
   });
+
+export const useListProjectMilestonesQuery = (params: ListProjectMilestonesRequest) => {
```
Comment (Daniilmass) [inline src/hooks/api-queries.ts:130]: Hmm, maybe we should consider adding docs to those exported consts as well? What do you think?

File: src/routes/projects_.$projectId.schedule.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  const renderGanttChart = () => {
+    return (
+      <Box sx={{ width: "auto", padding: 0 }} p={2}>
+        <Typography variant="body1">Chart placeholder content</Typography>
+      </Box>
+    );
+  };
+
+  return (
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.schedule.tsx:122]: Add TSdocs

File: src/routes/projects_.$projectId.schedule.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+import { createFileRoute } from "@tanstack/react-router";
+import { FlexColumnLayout } from "components/generic/flex-column-layout";
+import LoadingTableCell from "components/generic/loading-table-cell";
+import NewMilestoneDialog from "components/milestones/new-milestone-dialog";
+import { useListProjectMilestonesQuery } from "hooks/api-queries";
+import { useTranslation } from "react-i18next";
+import { DateTime } from "luxon";
+import ProgressBadge from "components/generic/progress-badge";
+
+export const Route = createFileRoute("/projects/$projectId/schedule")({
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.schedule.tsx:25]: Consider adding TSdocs

File: src/routes/projects_.$projectId.schedule.tsx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  component: ScheduleIndexRoute,
+});
+
+function ScheduleIndexRoute() {
+  const { t } = useTranslation();
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.schedule.tsx:30]: Consider adding TSdocs

File: src/routes/projects_.$projectId.schedule.tsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+          {t("scheduleScreen.objectives")}
+        </Typography>
+        <Box sx={{ display: "flex", flexDirection: "row" }}>
+          {renderProjectMilestonesTable()}
+          {renderGanttChart()}
+        </Box>
+      </Card>
+    </FlexColumnLayout>
+  );
+}
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.schedule.tsx:143]:
```suggestion
};
```

File: src/routes/projects_.$projectId.schedule.tsx [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+            <TableBody>{renderProjectMilestonesRows()}</TableBody>
+          </Table>
+        </TableContainer>
+      </Box>
+    );
+  };
+
+  /**
+   * Renders the milestone Gantt chart
+   */
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.schedule.tsx:113]:
```suggestion
   * Renders the milestone Gantt chart
   *
   * TODO: implement a gantt chart
   */
```

File: src/routes/projects_.$projectId.tracking.tsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { createFileRoute } from "@tanstack/react-router";
+
+export const Route = createFileRoute("/projects/$projectId/tracking")({
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.tracking.tsx:3]: Consider adding TSdocs

File: src/routes/projects_.$projectId.tracking.tsx [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+  component: TrackingIndexRoute,
+});
+
+function TrackingIndexRoute() {
```
Comment (Daniilmass) [inline src/routes/projects_.$projectId.tracking.tsx:7]: Consider adding TSdocs

**Review by Daniilmass — CHANGES_REQUESTED** [review]

File: src/components/milestones/new-milestone-dialog.tsx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+  const queryClient = useQueryClient();
+  const { projectId } = Route.useParams();
+  const [open, setOpen] = useState(false);
+  const [milestoneData, setMilestoneData] = useState<MilestoneFormData>({
+    name: "",
+    startDate: null,
+    endDate: null,
+  });
+
+  const createMilestoneMutation = useMutation({
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/milestones/new-milestone-dialog.tsx:31]: I'm not sure, other mutations in the project don't have them but I'm happy to add TSDocs for them all if we think its worth it?

File: src/hooks/api-queries.ts (hunk 4 above)
Comment (Jdallos) [reply src/hooks/api-queries.ts:130]: I agree.

File: src/routes/projects_.$projectId.schedule.tsx (hunk 8 above)
Comment (Jdallos) [reply src/routes/projects_.$projectId.schedule.tsx:143]: The linter is configured to remove semi-colon at the end of the file

File: src/components/milestones/new-milestone-dialog.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/milestones/new-milestone-dialog.tsx]: Considering other places we added docs I have added them to all mutation also

[the remark this answers is not in the document]
Comment (Daniilmass) [reply src/components/milestones/new-milestone-dialog.tsx]: If there is a sane number of functions using useMutation - I think it is worth adding docs to them

File: src/components/milestones/new-milestone-dialog.tsx (hunk 1 above)
Comment (Daniilmass) [reply src/components/milestones/new-milestone-dialog.tsx:85]: Loader was exactly what I meant

**Review by Daniilmass — APPROVED** [review]

**Pull request merged**